### PR TITLE
Script to work with all repos at once

### PR DESCRIPTION
**Pull request by szarnekow**
Usage: `./gitAll stash` or `./gitAll checkout -b myBranchName`

Also:
`./adjustPipelines.sh` resets the build to master branches
`./adjustPipelines.sh myBranch` redirects the build repos to the given branch name, if the repo does have such a branch

### CODE REVIEW

File: adjustPipelines.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+branchname=${1:-master}
+
+isBranch() {
+	git show-branch "$branchname" &> /dev/null
+}
+
+isMaster() {
+	test $branchname != master && isBranch
```
Comment (cdietrich) [inline adjustPipelines.sh]: ?

File: adjustPipelines.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			allDirectories | while read -r repo
+			do
+				repository=$(toDir $repo)
+				if [[ $directory != $repository ]]
+				then
+					logicalname=$(name $repo)
+					if changeDir $repository
+					then
+						sed -i '' "s%'http://services.typefox.io/open-source/jenkins/job/$logicalname/job/[^/]*/lastSuccessfulBuild/artifact/build/maven-repository/'%jenkinsPipelineRepo('$logicalname')%" $upstream
+						isMaster ||	sed -i '' "s%jenkinsPipelineRepo('$logicalname')%'http://services.typefox.io/open-source/jenkins/job/$logicalname/job/$1/lastSuccessfulBuild/artifact/build/maven-repository/'%" $upstream
```
Comment (cdietrich) [inline adjustPipelines.sh]: sometimes we use lastStableBuild.
i wonder if we simply can set -PupstreamBranch=xxxx somehow

File: adjustPipelines.sh (hunk 1 above)
Comment (szarnekow) [reply adjustPipelines.sh]: good catch

File: adjustPipelines.sh (hunk 2 above)
Comment (szarnekow) [reply adjustPipelines.sh]: theoretically yes, I was thinking about this, too. But that would really require all repos to be branched where the S&R allows to work with two out of six to be branched.

**Comment by cdietrich** [comment]
looks good

**Pull request merged**